sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/adapter/AbstractConnectionAdapter.java

```diff
@@ -204,7 +203,7 @@ public void execute(final Connection connection) throws SQLException {
     public final void close() throws SQLException {
         closed = true;
         MasterVisitedManager.clear();
-        TransactionTypeHolder.clear();
+        // TransactionTypeHolder.clear();
```

Comment:
TransactionTypeHolder should always clear after shardingConnection was closed by user.